### PR TITLE
[MRG+1] DOC Update semilogy docstring to numpy doc format

### PR DESCRIPTION
**Pull request by JuliaSprenger**
Hi Matplotlib Team,
here is an update of the semilogy docstring to the new numpy doc format.
Thanks!
Julia Sprenger

@NelleV

### CODE REVIEW

**Comment by NelleV** [comment]
This looks good! Thanks for the patch.

@tacaswell This is ready for review (and for merge).

**Comment by NelleV** [comment]
There seems to be formatting problems in the PR I did not pick up in my first review.

line 1611  W291 trailing whitespace
line 1621 W291 trailing whitespace
line 1623 W291 trailing whitespace
line 1631 W293 blank line contains whitespace

File: lib/matplotlib/axes/_axes.py [hunk 1]
```diff
@@ -1595,41 +1595,40 @@ def semilogx(self, *args, **kwargs):
...
 
-        %(Line2D)s
+        Other Parameters
+        ----------------        
+        kwargs : `~matplotlib.lines.Line2D` properties,
+        `~matplotlib.pylab.plot` and 
+        `matplotlib.axes.Axes.set_yscale` arguments.
 
-        .. seealso::
+        %(Line2D)s
```
Comment (QuLogic) [inline lib/matplotlib/axes/_axes.py]: `Line2D` probably also needs to be updated to get this interpolation correct.

File: lib/matplotlib/axes/_axes.py [hunk 2]
```diff
@@ -1595,41 +1595,40 @@ def semilogx(self, *args, **kwargs):
...
-        :class:`~matplotlib.lines.Line2D` properties:
+        Returns
+        -------
+        `~matplotlib.lines.Line2D`
+            Line instance of the plot.
 
-        %(Line2D)s
+        Other Parameters
+        ----------------        
+        kwargs : `~matplotlib.lines.Line2D` properties,
```
Comment (QuLogic) [inline lib/matplotlib/axes/_axes.py]: It's not `kwargs`; it's `**kwargs` (which should be documented as `\*\*kwargs` or enclosed in double-backticks (which I can't figure out how to use on GitHub without triggering the code styling.))

**Comment by QuLogic** [comment]
It appears you have not [set the config for your git client](https://git-scm.com/book/en/v2/Getting-Started-First-Time-Git-Setup#Your-Identity). All your commits are attributed to "Your Name <you@example.com>", which is probably not what you want.

**Comment by JuliaSprenger** [comment]
I fixed the trailing whitespaces, but I can't figure out why there are still failing tests.
Regarding the git configuration: I would leave it this way for now and do a better job next time.

**Comment by NelleV** [comment]
Hi @JuliaSprenger 
The failing test is not related to your PR.
Thanks for the contribution! Here is how it looks:

semilogy

**Comment by QuLogic** [comment]
The name on the first commit needs to be fixed. If you are not comfortable with doing so before hand, we need to remember to do so before merging.

**Comment by tacaswell** [comment]
I am not super concerned about semi-bogus names on commits.  We have not been strict about that in the past and we do not have any need for strict contribution audits.

**Comment by QuLogic** [comment]
It's fine if it used something unique from the computer or similar, but this is a generic name from git. It cannot be fixed by `.mailmap`.

**Comment by JuliaSprenger** [comment]
I changed the name and email for the first commit using rebase. I hope nothing else changed, but it seems to work.

**Comment by NelleV** [comment]
Thanks!

**Comment by tacaswell** [comment]
backported to v2.x as 873d393

**Comment by tacaswell** [comment]
Thanks @JuliaSprenger!

 Hopefully we will hear from you again soon.

**Pull request merged**